core/src/main/java/org/apache/calcite/sql/SqlDataTypeSpec.java

```diff
@@ -162,7 +161,7 @@ public TimeZone getTimeZone() {
   /** Returns a copy of this data type specification with a given
    * nullability. */
   public SqlDataTypeSpec withNullable(Boolean nullable) {
-    if (SqlFunctions.eq(nullable, this.nullable)) {
+    if (nullable.equals(this.nullable)) {
```

Comment:
SqlFunctions.eq() assumes neither side may be null, but this.nullable may be null, so assumption is broken here, and throwing NPE with modified implementation. Instead of handling null(s) in SqlFunctions.eq() I just modified this line. Is it OK?